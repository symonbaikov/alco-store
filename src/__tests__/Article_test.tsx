import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ArticleList from '../components/Article/Article';

describe('ArticleList', () => {
  it('renders correctly with all props', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ArticleList
          id="johnnie-walker"
          title="article.johnnie_walker.title"
          date="14 января 2021"
          image="/images/a7svkjdjilymj8d3gbw4syso5w1k9vcb.jpg"
          description="article.johnnie_walker.desc"
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with another article', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ArticleList
          id="kvint"
          title="article.kvint.title"
          date="22 сентября 2020"
          image="/images/5cv4mhwhzjwq02m6449mgqlu05g65e0p.jpg"
          description="article.kvint.desc"
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
}); 