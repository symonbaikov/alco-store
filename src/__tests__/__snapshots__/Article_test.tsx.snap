// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly with all props 1`] = `
<DocumentFragment>
  <div
    class="articles-wrapper"
  >
    <div
      class="article-card"
    >
      <a
        data-discover="true"
        href="/article/johnnie-walker"
      >
        <img
          alt="article.johnnie_walker.title"
          class="article-image"
          src="/images/a7svkjdjilymj8d3gbw4syso5w1k9vcb.jpg"
        />
        <div
          class="article-info"
        >
          <div
            class="article-date"
          >
            14 января 2021
          </div>
          <div
            class="article-title"
          >
            article.johnnie_walker.title
          </div>
          <div
            class="article-desc"
          >
            article.johnnie_walker.desc
          </div>
        </div>
      </a>
    </div>
    <div
      class="article-card"
    >
      <a
        data-discover="true"
        href="/article/merlot-vs-cabernet"
      >
        <img
          alt="article.merlot_vs_cabernet.title"
          class="article-image"
          src="/images/q09x5tclzqv9yhc5c5ow1kr943sw1yeu.jpg"
        />
        <div
          class="article-info"
        >
          <div
            class="article-date"
          >
            30 сентября 2020
          </div>
          <div
            class="article-title"
          >
            article.merlot_vs_cabernet.title
          </div>
          <div
            class="article-desc"
          >
            article.merlot_vs_cabernet.desc
          </div>
        </div>
      </a>
    </div>
    <div
      class="article-card"
    >
      <a
        data-discover="true"
        href="/article/wine-and-chocolate"
      >
        <img
          alt="article.wine_and_chocolate.title"
          class="article-image"
          src="/images/f0npxgt2j198vhcc6iyzlokeblismj5h.jpg"
        />
        <div
          class="article-info"
        >
          <div
            class="article-date"
          >
            26 сентября 2020
          </div>
          <div
            class="article-title"
          >
            article.wine_and_chocolate.title
          </div>
          <div
            class="article-desc"
          >
            article.wine_and_chocolate.desc
          </div>
        </div>
      </a>
    </div>
    <div
      class="article-card"
    >
      <a
        data-discover="true"
        href="/article/courvoisier"
      >
        <img
          alt="article.courvoisier.title"
          class="article-image"
          src="/images/1oxjo496yfmiyc4wih6micguzx0joddr.jpg"
        />
        <div
          class="article-info"
        >
          <div
            class="article-date"
          >
            24 сентября 2020
          </div>
          <div
            class="article-title"
          >
            article.courvoisier.title
          </div>
          <div
            class="article-desc"
          >
            article.courvoisier.desc
          </div>
        </div>
      </a>
    </div>
    <div
      class="article-card"
    >
      <a
        data-discover="true"
        href="/article/kvint"
      >
        <img
          alt="article.kvint.title"
          class="article-image"
          src="/images/5cv4mhwhzjwq02m6449mgqlu05g65e0p.jpg"
        />
        <div
          class="article-info"
        >
          <div
            class="article-date"
          >
            22 сентября 2020
          </div>
          <div
            class="article-title"
          >
            article.kvint.title
          </div>
          <div
            class="article-desc"
          >
            article.kvint.desc
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ArticleList renders correctly with another article 1`] = `
<DocumentFragment>
  <div
    class="articles-wrapper"
  >
    <div
      class="article-card"
    >
      <a
        data-discover="true"
        href="/article/johnnie-walker"
      >
        <img
          alt="article.johnnie_walker.title"
          class="article-image"
          src="/images/a7svkjdjilymj8d3gbw4syso5w1k9vcb.jpg"
        />
        <div
          class="article-info"
        >
          <div
            class="article-date"
          >
            14 января 2021
          </div>
          <div
            class="article-title"
          >
            article.johnnie_walker.title
          </div>
          <div
            class="article-desc"
          >
            article.johnnie_walker.desc
          </div>
        </div>
      </a>
    </div>
    <div
      class="article-card"
    >
      <a
        data-discover="true"
        href="/article/merlot-vs-cabernet"
      >
        <img
          alt="article.merlot_vs_cabernet.title"
          class="article-image"
          src="/images/q09x5tclzqv9yhc5c5ow1kr943sw1yeu.jpg"
        />
        <div
          class="article-info"
        >
          <div
            class="article-date"
          >
            30 сентября 2020
          </div>
          <div
            class="article-title"
          >
            article.merlot_vs_cabernet.title
          </div>
          <div
            class="article-desc"
          >
            article.merlot_vs_cabernet.desc
          </div>
        </div>
      </a>
    </div>
    <div
      class="article-card"
    >
      <a
        data-discover="true"
        href="/article/wine-and-chocolate"
      >
        <img
          alt="article.wine_and_chocolate.title"
          class="article-image"
          src="/images/f0npxgt2j198vhcc6iyzlokeblismj5h.jpg"
        />
        <div
          class="article-info"
        >
          <div
            class="article-date"
          >
            26 сентября 2020
          </div>
          <div
            class="article-title"
          >
            article.wine_and_chocolate.title
          </div>
          <div
            class="article-desc"
          >
            article.wine_and_chocolate.desc
          </div>
        </div>
      </a>
    </div>
    <div
      class="article-card"
    >
      <a
        data-discover="true"
        href="/article/courvoisier"
      >
        <img
          alt="article.courvoisier.title"
          class="article-image"
          src="/images/1oxjo496yfmiyc4wih6micguzx0joddr.jpg"
        />
        <div
          class="article-info"
        >
          <div
            class="article-date"
          >
            24 сентября 2020
          </div>
          <div
            class="article-title"
          >
            article.courvoisier.title
          </div>
          <div
            class="article-desc"
          >
            article.courvoisier.desc
          </div>
        </div>
      </a>
    </div>
    <div
      class="article-card"
    >
      <a
        data-discover="true"
        href="/article/kvint"
      >
        <img
          alt="article.kvint.title"
          class="article-image"
          src="/images/5cv4mhwhzjwq02m6449mgqlu05g65e0p.jpg"
        />
        <div
          class="article-info"
        >
          <div
            class="article-date"
          >
            22 сентября 2020
          </div>
          <div
            class="article-title"
          >
            article.kvint.title
          </div>
          <div
            class="article-desc"
          >
            article.kvint.desc
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
